csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -164,30 +164,27 @@ public TBuilder Clear()
 
         public ArrowBuffer ValueBuffer => Data.Buffers[2];
 
-        public ReadOnlySpan<int> ValueOffsets => ValueOffsetsBuffer.Span.CastTo<int>().Slice(0, Length + 1);
+        public ReadOnlySpan<int> ValueOffsets => ValueOffsetsBuffer.Span.CastTo<int>().Slice(Offset, Length + 1);
 
         public ReadOnlySpan<byte> Values => ValueBuffer.Span.CastTo<byte>();
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetValueOffset(int index)
         {
-            return ValueOffsets[Offset + index];
+            return ValueOffsets[index];
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetValueLength(int index)
         {
-            var offsets = ValueOffsets;
-            var offset = Offset + index;
-
-            return offsets[offset + 1] - offsets[offset];
+            return ValueOffsets[index + 1] - ValueOffsets[index];
```

Comment:
Do we need to check the index for out of range?
Also - no need to call the property twice. Can cache the span in a local variable.

Reply:
Thank you for your comments.

- I added a boundary check to this method and similar methods.
- I modified to use local variable.